src/boot/efi/stub.c

```diff
@@ -26,19 +26,30 @@ DECLARE_NOALLOC_SECTION(".sdmagic", "#### LoaderInfo: systemd-stub " GIT_VERSION
 
 DECLARE_SBAT(SBAT_STUB_SECTION_TEXT);
 
-static EFI_STATUS combine_initrd(
-                EFI_PHYSICAL_ADDRESS initrd_base, size_t initrd_size,
+/* Combine initrds by concatenation in memory.
+ * uki_initrds come from sections in the UKI and will be padded,
+ * extra initrds are generated and don't need further padding */
+static EFI_STATUS combine_initrds(
+                EFI_PHYSICAL_ADDRESS uki_initrds[], const size_t uki_initrd_sizes[], size_t n_uki_initrds,
```

Comment:
I wouldn't go too fancy here and just add two new arguments to specify microcode base and size and use `0` for no microcode initrd specified.

Reply:
You can still treat both the microcode and the actual initrd as an array inside the function though.

Reply:
Sure, can do. I figured this makes it a bit more mergeable if #28070 (multiple initrds) want to move forward at some point, but I can keep it simple for now